azurerm/internal/services/network/tests/resource_arm_express_route_gateway_test.go

```diff
@@ -0,0 +1,212 @@
+		},
+	})
+}
+
+func testCheckAzureRMExpressRouteGatewayExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.ExpressRouteGatewaysClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("ExpressRoute Gateway not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: ExpressRoute Gateway %q (Resource Group %q) does not exist", name, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on network.ExpressRouteGatewaysClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMExpressRouteGatewayDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Network.ExpressRouteGatewaysClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_express_route_gateway" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on network.ExpressRouteGatewaysClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMExpressRouteGateway_basic(data acceptance.TestData) string {
+	template := testAccAzureRMExpressRouteGateway_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_gateway" "test" {
+  name                = "acctestER-gateway-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  virtual_hub_id      = azurerm_virtual_hub.test.id
+  scale_units         = 1
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMExpressRouteGateway_complete(data acceptance.TestData, scaleUnits int) string {
+	template := testAccAzureRMExpressRouteGateway_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_gateway" "test" {
+  name                = "acctestER-gateway-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  virtual_hub_id      = azurerm_virtual_hub.test.id
+  scale_units         = %d
+
+  tags = {
+    Hello = "World"
+  }
+}
+`, template, data.RandomInteger, scaleUnits)
+}
+
+func testAccAzureRMExpressRouteGateway_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMExpressRouteGateway_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_gateway" "test" {
```

Comment:
this needs to be a unique resource name:

```suggestion
resource "azurerm_express_route_gateway" "import" {
```